ampersand-state.js

```diff
@@ -177,11 +178,12 @@ assign(Base.prototype, Events, {
             }
 
             isEqual = this._getCompareForType(def.type);
+            onChange = this._getOnChangeForType(def.type);
             dataType = this._dataTypes[def.type];
 
             // check type if we have one
             if (dataType && dataType.set) {
-                cast = dataType.set(newVal);
+                cast = dataType.set.call(this, newVal, options);
```

Comment:
AFAIK the only built-in type using `this` inside of `set` was 'state', but I would have to check to be sure. I think ampersand-view defines some datatypes too, not sure about other modules.